### PR TITLE
Advanced isolated network egress destination cidr support added

### PR DESCRIPTION
**Pull request by jayapalu**
Added the destination cidr support for the advanced zone isolated network egress rule.
With this now user can control both source and destination ip address of the egress firewall rules.

### CODE REVIEW

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 355
 Hypervisor xenserver
 NetworkType Advanced
 Passed=78
 Failed=17
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_vm_snapshots.py

 * test_change_service_offering_for_vm_with_snapshots Failed

 * test_01_create_vm_snapshots Failed

 * test_02_revert_vm_snapshots Failed

 * test_03_delete_vm_snapshots Failed

* test_service_offerings.py

 * test_04_change_offering_small Failed

* test_loadbalance.py

 * test_01_create_lb_rule_src_nat Failed

 * test_02_create_lb_rule_non_nat Failed

 * test_assign_and_removal_lb Failed

* test_router_dns.py

 * test_router_dns_guestipquery Failed

* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failing since 2 runs

* test_volumes.py

 * test_01_create_volume Failed

 * test_02_attach_volume Failed

* test_vm_life_cycle.py

 * test_10_attachAndDetach_iso Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 2 runs


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 387
 Hypervisor xenserver
 NetworkType Advanced
 Passed=100
 Failed=5
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed

 * test_03_RVR_Network_check_router_state Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by jayapalu** [comment]
I tested manually test_02_isolate_network_FW_PF_default_routes_egress_false Failed. It worked for me.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 409
 Hypervisor xenserver
 NetworkType Advanced
 Passed=100
 Failed=5
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed

 * test_03_RVR_Network_check_router_state Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

File: tools/appliance/definitions/systemvmtemplate/install_systemvm_packages.sh [hunk 1]
```diff
@@ -67,6 +67,8 @@ function install_packages() {
     xenstore-utils libxenstore3.0 \
     conntrackd ipvsadm libnetfilter-conntrack3 libnl-3-200 libnl-genl-3-200 \
     ipcalc \
+    ipset \
+    openjdk-7-jre-headless \
```
Comment (kishankavala) [inline tools/appliance/definitions/systemvmtemplate/install_systemvm_packages.sh]: Only ipset is required here

File: engine/schema/src/com/cloud/upgrade/dao/Upgrade4920to41000.java [hunk 2]
```diff
@@ -53,6 +55,7 @@ public boolean supportsRollingUpgrade() {
 
     @Override
     public void performDataMigration(Connection conn) {
+        updateSourceCidrs(conn);
```
Comment (kishankavala) [inline engine/schema/src/com/cloud/upgrade/dao/Upgrade4920to41000.java]: Please add comment on why source cidr is updated

Comment (jayapalu) [inline engine/schema/src/com/cloud/upgrade/dao/Upgrade4920to41000.java]: Updated

**Comment by jayapalu** [comment]
tag:mergeready

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 415
 Hypervisor xenserver
 NetworkType Advanced
 Passed=100
 Failed=5
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed

 * test_03_RVR_Network_check_router_state Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by jayapalu** [comment]
This PR has template change (ipset command installation). So while testing this PR we need to generate new template and use it.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 677
 Hypervisor xenserver
 NetworkType Advanced
 Passed=108
 Failed=4
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 2 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by swill** [comment]
Do we know why the egress tests are failing. There are a lot of network tests failing in this PR. I don't think this is ready to merge without another run of the CI tests to clarify if there are still network failures.

**Comment by jayapalu** [comment]
@swill I have fixed issue for the failure of above 4 test cases. This PR has the template change (ipset installation). CI need to run  by generating template on this PR and use the same.

**Comment by jayapalu** [comment]
I have tested the below test test manually and with automation test. Those are passing working now.
Please find one test result below.

Some time the test failures are observed in RVR because of the RVR issues.  If Some one is want to test this, either seed the template with ipset installed or bring up the VR and install the ipset command and test it.

test_01_isolate_network_FW_PF_default_routes_egress_true Failed
test_02_isolate_network_FW_PF_default_routes_egress_false Failed
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 2 runs

$ nosetests-2.7 --with-marvin --marvin-config=/Users/jayapal_uradi/dev/advanced.cfg --with-xunit  --xunit-file=/tmp/test/testslog /Users/jayapal_uradi/dev/github/cloudstack/test/integration/smoke/test_routers_network_ops.py -a tags=jayapal --zone=zone1 --hypervisor=xenserver

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /tmp//MarvinLogs//May_12_2017_16_00_03_IL3AHU. All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
Starting test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false...
Creating Network Offering with default egress FALSE
Creating network with network offering: f0ea499e-82aa-4b54-aad3-594ddfed4ca4
===final results are now copied to: /tmp//MarvinLogs/test_routers_network_ops_OXMLIP===
HSL007948:cloudstack jayapal_uradi$ vi /tmp//MarvinLogs/test_routers_network_ops_OXMLIP/results.txt 
HSL007948:cloudstack jayapal_uradi$ nosetests-2.7 --with-marvin --marvin-config=/Users/jayapal_uradi/dev/advanced.cfg --with-xunit  --xunit-file=/tmp/test/testslog /Users/jayapal_uradi/dev/github/cloudstack/test/integration/smoke/test_routers_network_ops.py -a tags=jayapal --zone=zone1 --hypervisor=xenserver

==== Marvin Init Started ====

=== Marvin Parse Config Successful ===

=== Marvin Setting TestData Successful===

==== Log Folder Path: /tmp//MarvinLogs//May_12_2017_16_05_13_EDJD08. All logs will be available here ====

=== Marvin Init Logging Successful===

==== Marvin Init Successful ====
Starting test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false...
Creating Network Offering with default egress FALSE
Creating network with network offering: ade3a83d-71d0-4085-be9a-ec43964d23d3
Created network with ID: 303c7b2e-d228-45e2-919e-ec364c93dcfd
Deploying VM in account: test-a-TestIsolatedNetworks-YGWH9M
Deployed VM in network: 303c7b2e-d228-45e2-919e-ec364c93dcfd
Listing routers for network: Test Network
Creating Firewall rule for VM ID: 79f5c3e9-457e-4154-8c34-e72dec5eba63


Creating NAT rule for VM ID: 79f5c3e9-457e-4154-8c34-e72dec5eba63
====Trying SSH Connection: Host:10.147.46.108 User:root                                   Port:22 RetryCnt:5===
===SSH to Host 10.147.46.108 port : 22 SUCCESSFUL===
{Cmd: ping -c 3 8.8.8.8 via Host: 10.147.46.108} {returns: [u'PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.', u'', u'--- 8.8.8.8 ping statistics ---', u'3 packets transmitted, 0 received, 100% packet loss, time 2009ms', u'']}
{Cmd: wget -t 1 -T 1 www.google.com via Host: 10.147.46.108} {returns: [u'--2017-05-12 10:41:09--  http://www.google.com/', u'Resolving www.google.com... 216.58.199.164, 2404:6800:4009:807::2004', u'Connecting to www.google.com|216.58.199.164|:80... failed: Connection timed out.', u'Connecting to www.google.com|2404:6800:4009:807::2004|:80... failed: Address family not supported by protocol.']}
{Cmd: wget -t 1 -T 5 www.google.com via Host: 10.147.46.108} {returns: [u'--2017-05-12 10:41:31--  http://www.google.com/', u'Resolving www.google.com... 216.58.199.164, 2404:6800:4009:807::2004', u'Connecting to www.google.com|216.58.199.164|:80... connected.', u'HTTP request sent, awaiting response... 302 Found', u'Location: http://www.google.co.in/?gfe_rd=cr&ei=W5EVWfWgEbPT8gfOqqHoAw [following]', u'--2017-05-12 10:41:31--  http://www.google.co.in/?gfe_rd=cr&ei=W5EVWfWgEbPT8gfOqqHoAw', u'Resolving www.google.co.in... 172.217.26.227, 2404:6800:4009:805::2003', u'Connecting to www.google.co.in|172.217.26.227|:80... connected.', u'HTTP request sent, awaiting response... 200 OK', u'Length: unspecified [text/html]', u"Saving to: `index.html?gfe_rd=cr&ei=W5EVWfWgEbPT8gfOqqHoAw'", u'', u'     0K .......... ..                                          1.66M=0.008s', u'', u"2017-05-12 10:41:31 (1.66 MB/s) - `index.html?gfe_rd=cr&ei=W5EVWfWgEbPT8gfOqqHoAw' saved [13167]", u'']}
===final results are now copied to: /tmp//MarvinLogs/test_routers_network_ops_60G2JL===
HSL007948:cloudstack jayapal_uradi$ 
HSL007948:cloudstack jayapal_uradi$ 
HSL007948:cloudstack jayapal_uradi$ cat /tmp//MarvinLogs/test_routers_network_ops_60G2JL/results.txt 
Test redundant router internals ... === TestName: test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 377.896s

OK

**Comment by jayapalu** [comment]
The failed test cases in CI results I have fixed and tested locally with marvin tests as well as manually  in my  setup.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 703
 Hypervisor xenserver
 NetworkType Advanced
 Passed=107
 Failed=5
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_loadbalance.py

 * test_01_create_lb_rule_src_nat Failed

 * test_02_create_lb_rule_non_nat Failed

 * test_assign_and_removal_lb Failed

* test_network.py

 * test_public_ip_admin_account Failed

* test_deploy_vm_root_resize.py

 * test_00_deploy_vm_root_resize Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Comment by jayapalu** [comment]
Tests are passing manually in my setup. It seems in CI test are failures are intermittent. 
HSL007948:cloudstack jayapal_uradi$ cat /tmp//MarvinLogs/test_loadbalance_7WPP1B/results.txt 
Test to create Load balancing rule with source NAT ... === TestName: test_01_create_lb_rule_src_nat | Status : SUCCESS ===
ok
Test to create Load balancing rule with non source NAT ... === TestName: test_02_create_lb_rule_non_nat | Status : SUCCESS ===
ok
Test for assign & removing load balancing rule ... === TestName: test_assign_and_removal_lb | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 3 tests in 883.903s

OK

HSL007948:cloudstack jayapal_uradi$ cat  /tmp//MarvinLogs/test_network_OAY0RJ/results.txt 
Test for Associate/Disassociate public IP address for admin account ... === TestName: test_public_ip_admin_account | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 72.054s

OK

**Comment by jayapalu** [comment]
smoke/test_over_provisioning smoke/test_password_server  is failing. It seems there issue with the CI. I tried triggering travis CI 3 times.

**Comment by jayapalu** [comment]
For CI intermittent failures I have posted manual results. So tagging merge ready.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 705
 Hypervisor xenserver
 NetworkType Advanced
 Passed=106
 Failed=6
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_routers.py

 * test_04_restart_network_wo_cleanup Failed

* test_ssvm.py

 * test_08_reboot_cpvm Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failed

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by bvbharatk** [comment]
Hi,
 CI ran again without the template changes. The results with template changes are  below.

ACS CI BVT Run
Sumarry:
Build Number 703
Hypervisor xenserver
NetworkType Advanced
Passed=107
Failed=5
Skipped=12

Link to logs Folder (search by build_no): https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

Failed tests:

test_loadbalance.py

test_01_create_lb_rule_src_nat Failed

test_02_create_lb_rule_non_nat Failed

test_assign_and_removal_lb Failed

test_network.py

test_public_ip_admin_account Failed

test_deploy_vm_root_resize.py

test_00_deploy_vm_root_resize Failed

Skipped tests:
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

Passed test suits:
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_routers_network_ops.py
test_disk_offerings.py

**Pull request merged**